src/asmcnc/job/job_data.py

```diff
@@ -422,14 +422,12 @@ def read_from_recovery_file(self):
             print("Could not read recovery info")
             print(str(traceback.format_exc()))
 
-    def write_to_recovery_file_after_cancel(self, cancel_line):
+    def write_to_recovery_file_after_cancel(self, cancel_line, cancel_time):
         try:
             # Account for number of lines added in by the software when running file
             cancel_line -= self.job_recovery_offset
 
-            # If job was cancelled before it started, no need to store recovery info
-            if cancel_line >= 0:
-
+            def write_to_file(cancel_line):
```

Comment:
I think move the function definition out of the try-except clause (so above the try:), as any failure at run time will still be caught, and it makes everything a little more readable.